compiler/parser/chapel.lex

```diff
@@ -154,7 +154,9 @@ param            return processToken(yyscanner, TPARAM);
 zip              return processToken(yyscanner, TZIP);
 pragma           return processToken(yyscanner, TPRAGMA);
 __primitive      return processToken(yyscanner, TPRIMITIVE);
+private          return processToken(yyscanner, TPRIVATE);
 proc             return processToken(yyscanner, TPROC);
+public           return processToken(yyscanner, TPUBLIC);
 record           return processToken(yyscanner, TRECORD);
 reduce           return processToken(yyscanner, TREDUCE);
```

Comment:
Yeah, it was a conscious choice.  I'll make sure that they are added as keywords to the spec when I add relevant descriptions of what they mean in context (should be soon).

Sure, I can fix the alphabetization of zip!